src/goto-diff/goto_diff_base.cpp

```diff
@@ -11,95 +11,134 @@ Author: Peter Schrammel
-        (json_result["newFunctions"].make_array(), new_functions);
-      convert_function_group(
-        json_result["modifiedFunctions"].make_array(), modified_functions);
-      convert_function_group(
-        json_result["deletedFunctions"].make_array(), deleted_functions);
-      out << ",\n" << json_result;
+      convert_function_group_json(
+        json_result["newFunctions"].make_array(),
+        new_functions, goto_model2);
+      convert_function_group_json(
+        json_result["modifiedFunctions"].make_array(),
+        modified_functions, goto_model2);
+      convert_function_group_json(
+        json_result["deletedFunctions"].make_array(),
+        deleted_functions, goto_model1);
+      result() << json_result;
       break;
     }
     case ui_message_handlert::uit::XML_UI:
     {
-      out << "not supported yet";
+      error() << "not supported yet" << eom;
     }
   }
-  return out;
 }
 
-void goto_difft::convert_function_group(
-  json_arrayt &result,
-  const irep_id_sett &function_group) const
+/// Output group of functions in plain text format
+/// \param group_name: the name of the group, e.g. "modified functions"
+/// \param function_group: set of function ids in the group
+/// \param goto_model: the goto model
+void goto_difft::output_function_group(
+  const std::string &group_name,
+  const irep_id_sett &function_group,
+  const goto_modelt &goto_model) const
 {
-  for(irep_id_sett::const_iterator it=function_group.begin();
-      it!=function_group.end(); ++it)
+  result() << group_name << ":\n";
+  for(const auto &function_name : function_group)
   {
-    convert_function(result.push_back(jsont()).make_object(), *it);
+    output_function(function_name, goto_model);
   }
 }
 
-void goto_difft::convert_function(
-  json_objectt &result,
-  const irep_idt &function_name) const
+/// Output function information in plain text format
+/// \param function_name: the function id
+/// \param goto_model: the goto model
+void goto_difft::output_function(
+  const irep_idt &function_name,
+  const goto_modelt &goto_model) const
 {
-  // the function may be in goto_model1 or goto_model2
-  if(
-    goto_model1.goto_functions.function_map.find(function_name) !=
-    goto_model1.goto_functions.function_map.end())
+  const auto goto_function_it =
+    goto_model.goto_functions.function_map.find(function_name);
+  PRECONDITION(goto_function_it != goto_model.goto_functions.function_map.end());
+
+  namespacet ns(goto_model.symbol_table);
+  const symbolt &symbol = ns.lookup(function_name);
+  result() << "  " << symbol.location.get_file() << ": " << function_name << "\n";
+  if(options.get_bool_option("show-properties"))
   {
-    const symbolt &symbol =
-      namespacet(goto_model1.symbol_table).lookup(function_name);
-    result["sourceLocation"] = json(symbol.location);
+    const goto_programt &goto_program = goto_function_it->second.body;
+      for(const auto &ins : goto_program.instructions)
```

Comment:
The `for` loop seems to be indented more than the  preceding line? Also: may I suggest to move the `const auto goto_function_it = ...` in here as it is not necessary for the other output just above?